examples/tactics/Normalization.fst

```diff
@@ -0,0 +1,50 @@
+module Normalization
+
+open FStar.Tactics
+
+let normalize (#t:Type) (x:t) : tactic unit =
+  dup;;
+  exact (quote x);;
+  norm [Delta; Primops];;
+  trefl
+
+let add_1 (x:int) : int = x + 1
+
+// add_2 is defined to be x + 1 + 1
+let add_2 (x:int) : int = synth_by_tactic (normalize (add_1 (add_1 x)))
+
+let four : int = synth_by_tactic (normalize (add_2 2))
+
+let _ = assert (four == 4)
```

Comment:
Done with a brand new tactic feature of looking up definitions (previously this only worked for definition of inductives). I also added it to regressions. If you're happy with the result, I can merge it into my branch, and then to master when we unfreeze. Thanks a lot for file!